### DISCUSSION TITLE
검색 엔진 최적화(SEO)에 대한 경험이 있나요?

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#6] [프론트 기술 질문 공간]
아시는 만큼 설명해주세요.

**Comment by javascriptbiggosoo** [comment]
시맨틱 마크업을 사용해 의미있는 컴포넌트를 구성하거나 img 태그의 alt 등과 같은 대체 정보에 신경을 쓰는 방식을 알고 있습니다.
리액트의 경우 페이지가 유저쪽에서 렌더링되기 때문에 SEO에서 불이익이 있고, Next와 같은 SSR 도구를 활용해 문제를 해결할 수 있다고 들었습니다만 아직 Next는 다뤄보지 못했습니다.

**Comment by michoball** [comment]
CSR 의 단점 중 하나가 SEO가 불리하다는 것입니다. 대표적인 CSR인 React로 만 개발을 거의 다 진행을 했기에 따로 SEO를 고려해서 개발을 진행하지는 않았는데요. 

실제 적용해본적은 없지만 알고있는 방법으로는 react-helmet 라이브러리를 사용해서 meta 데이터를 JSX 안에 넣어주는 방법과  react-snap 라이브러리를 사용해서 미리 프로젝트내에 page를 돌아 SEO를 위한 데이터를 찾아 배포 페이지에 넣어주는 방법이 있을 것 같습니다.

아니면 SSR 이나 SSG를 가능하게 해주는 Next.js 를 쓰는 것도 방법일 것 같습니다.

**Comment by skyhanull** [comment]
SEO란 검색엔진 최적화를 의미합니다
csr을 이용하는 리엑트에서는 첫랜더링시 모든 데이터를 가져오지만 html이 빈칸으로 들어오기에 검색엔진에서 읽을 수 없습니다.
하지만 ssr을 사용하는 next.js는 페이지 렌더링시 지금 필요한 데이터만 들어오므로 html에 내용이 저장이 됩니다. html에 정보가 있으니 검색엔진에 검색이 가능해집니다.

**Comment by WooGie911** [comment]
SEO(Search Engine Optimization) 즉 검색 엔진 최적화는 웹사이트가 유기적인 검색 방식을 통해 검색 엔진에서 상위에 노출될 수 있도록 최적화하는 과정을 의미합니다. 
쉽게 접할수 있는 SEO 과정의 예시로는 링크를 공유할 때 볼 수 있는 OG(Open Gragh) 태그와 검색엔진에 검색했을 때 노출되는 메타(Meta) 태그 등이 있습니다. 메타 태그는 <title> 태그, <description> 태그, <robots> 태그, <canonical> 태그 등이 있습니다.

**Comment by bona373737** [comment]
검색엔진은 크롤링,인덱싱,랭킹 이 세단계를 거쳐 검색결과를 만들어냅니다. 
검색엔진최적화를 위해선 기본적으로 검색엔진이 웹페이지 내의 데이터를 잘 읽어올 수 있어야 하는데, CSR방식으로 구현된 웹페이지들은 여러개의 페이지가 존재하는 것처럼 보여지지만 내부적으로는 단 1개의 html파일만을 갖고 있기때문에 수집해올 수 있는 html정보에가 제한적인다. 반면 SSR방식으로 구현된 웹페이지들은 각기 다른 html파일로 구성되어 있기 때문에 각각의  html정보를 수집해 올 수 있다. 

*구글은 CSR방식의 웹사이트도 SEO가능..? 
*리택트 SEO개선방법..react-helment, pre-rendering..?
